neo/rawio/spikeglxrawio.py

```diff
@@ -172,17 +178,38 @@ def _get_analogsignal_chunk(self, block_index, seg_index, i_start, i_stop,
                                 stream_index, channel_indexes):
         stream_id = self.header['signal_streams'][stream_index]['id']
         memmap = self._memmaps[seg_index, stream_id]
-
         if channel_indexes is None:
-            channel_indexes = slice(channel_indexes)
-
-        if not isinstance(channel_indexes, slice):
+            if self.load_sync_channel:
+                channel_selection = slice(None)
+            else:
+                channel_selection = slice(-1)
+        elif isinstance(channel_indexes, slice):
+            if self.load_sync_channel:
+                # simple
+                channel_selection = channel_indexes
+            else:
+                # more tricky because negative
+                sl_start = channel_indexes.start
+                sl_stop = channel_indexes.stop
+                sl_step = channel_indexes.step
+                if sl_stop is not None and sl_stop < 0:
+                    sl_stop = sl_stop - 1
+                elif sl_stop is None:
```

Comment:
Shouldn't `sl_stop` stay `None` then?

Reply:
No because we remove the last channel in that case.